services/workflows-service/src/alert/alert.service.ts

```diff
@@ -140,8 +152,56 @@ export class AlertService {
     }
   }
 
-  // Specific alert check logic based on the definition
-  private async checkAlert(alertDefinition: AlertDefinition) {
+  async checkOngoingMonitoringAlert(businessReport: BusinessReport, businessCompanyName: string) {
+    const alertDefinitions = await this.alertDefinitionRepository.findMany(
+      {
+        where: {
+          enabled: true,
+          monitoringType: MonitoringType.ongoing_merchant_monitoring,
+        },
+      },
+      [businessReport.projectId],
+    );
+
+    const alertDefinitionsCheck = alertDefinitions.map(async alertDefinition => {
+      const alertResultData = await this.dataAnalyticsService.checkMerchantOngoingAlert(
+        businessReport,
+        (alertDefinition.inlineRule as InlineRule).options as CheckRiskScoreOptions,
+        alertDefinition.defaultSeverity,
+      );
+
+      if (alertResultData) {
+        const { reportId, id: businessReportId, businessId, projectId } = businessReport;
+        const subjects = { businessId, projectId };
+
+        const subjectArray = Object.entries(subjects).map(([key, value]) => ({
+          [key]: value,
+        }));
+
+        const createAlertReference = this.createAlert;
+
+        return [
+          alertDefinition,
+          subjectArray,
+          { subjectArray },
+          {
+            ...alertResultData,
+            reportId,
+            businessReportId,
+            businessCompanyName,
+          },
+        ] satisfies Parameters<typeof createAlertReference>;
+      }
+    });
+
+    const alertCreateArgs = (await Promise.all(alertDefinitionsCheck))
```

Comment:
Please replace with allSettled or add Try-Catch in your alert check fn

<img width="892" alt="image" src="https://github.com/ballerine-io/ballerine/assets/6435752/2bee44e4-d8e0-42ad-92af-9baee7823c8a">

Reply:
it's not as simple as it sounds. and there isn't much value as I want it to crush in case it fails.

therefore I prefer t to be promise.all ( also we're talking about a quick process of checking at most 5-6 validations )